lib/api/bucketPutVersioning.js

```diff
@@ -50,6 +56,16 @@ function _parseXML(request, log, cb) {
     });
 }
 
+function _checkBackendVersioningImplemented(bucket) {
+    const bucketLocation = bucket.getLocationConstraint();
+    const bucketLocationType = config.locationConstraints[bucketLocation].type;
```

Comment:
You can use: `getLocationConstraintType()`
https://github.com/scality/S3/blob/master/lib/Config.js#L797